cmd/fleetctl/fleetctl/api.go

```diff
@@ -264,17 +281,22 @@ func apiCommand() *cli.Command {
 			flField = c.StringSlice("F")
 			flHeader = c.StringSlice("H")
 
-			if len(flField) > 0 {
+			switch method {
+			case "GET":
 				for _, each := range flField {
```

Comment:
note for reviewers: the length check `if len(flField) > 0 {` was redundant.

an empty slice is nil, and it is safe to iterate on.